sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3258,7 +3258,9 @@ class SQLConf extends Serializable with Logging {
   }
 
   def clear(): Unit = {
+    val reserved = settings.asScala.filterKeys(staticConfKeys.contains).toMap
```

Comment:
does `staticConfKeys` include the core configs?

Reply:
it doesn't. If a user calls it, it will make the SparkConf in SQLConf all gone